feat(tools): add python script used for migrating to github issues

Swiftb0y opened the pull request:
@ywwg requested we include the source for the import tool in our source tree.

Swiftb0y committed:
feat(tools): add python script used for migrating to github issues

daschuer commented:
The pre-commit issue is s false positive.

daschuer reviewed (approved):
Thank you. 
@ywwg 

Do you want to have a look before merge? Do we want to merge the final version after it has been used or track changes this way?

Holzhaus commented on tools/json2github.py:
```diff
@@ -0,0 +1,666 @@
+    "johan-lasperas": "johanLsp",
+    "josepma": "JosepMaJAZ",
+    "juha-pitkanen": "JuhaPit",
+    "jus": "esbrandt",  # codespell:ignore jus
```
I line comments don't work, you need to add `jus` to the `.codestellignore` file.

Swiftb0y replied:
Yeah I'm aware. There is a PR that introduces inline ignores. I didn't want to add "jus" to the global `.codespellignore` because it could be a likely typo. I'd just wait for https://github.com/codespell-project/codespell/pull/2400 before merging. :shrug:

Holzhaus replied:
The alternative to waiting is to add the entire line to a `.codespellignorelines` file and adding `--exclude-file=.codespellignorelines` to the codespell command line. Then you don't have to worry about false negatives.

Swiftb0y commented:
I'd also wait with merging until we actually did the import. Its likely I'll still have to make some adjustments to the script.

Swiftb0y committed:
fix(tools): fix json2github.py not respecting RateLimit timeout

This was due to mistakes in the calculation for the wait time
which arose because of improper handling of timezones.

Swiftb0y committed:
fix(tools): json2github handle bugs outside mixxx

Issues can be duplicates of  of issues that don't belong to the mixxx
project. Since those are obviously not imported
to github, we link to their originals
on launchpad instead of github.

Swiftb0y committed:
feat(tools): tool to fix issue-json after transfer

Since issues get a new ID/number during bulk-transfer we need to
update our launchpad-export-json with the new numbers.
This is done via a mapping file we get from githubs bulk-transfer tools.

Swiftb0y committed:
feat(tools): json2github handle milestones fixup [WIP]

Swiftb0y committed:
improve log output when github ratelimit exceeded

Swiftb0y committed:
dedup code

Swiftb0y committed:
retry on connection errors

Swiftb0y committed:
improve logging in resume logic

Swiftb0y committed:
add backlinking script

Holzhaus commented on tools/launchpad2githubBacklinks.py:
```diff
@@ -0,0 +1,61 @@
+# /usr/bin/env python3
```
Can we stick to snake_case filenames instead of camelCase?

mYnDstrEAm commented:
Is this specific to mixxx? If not, create another separate repo for it so other Launchpad repos can find and use it too. Also see #10952

Holzhaus commented:
> Is this specific to mixxx? If not, create another separate repo for it so other Launchpad repos can find and use it too. Also see #10952

It is kind of specific because it hardcodes the term "Mixxx" in various locations, maps certain Launchpad usernames to GitHub usernames, etc.

I have no intention to spend any more time on this, but the code is open source and distributed under the terms of the GPL. Feel free to modify it and put into a separate repository if you want.

daschuer commented:
Do we want to merge this now? 
It is for documentation purpose only, so it should be safe.

Holzhaus commented:
The pre-commit checks fail. If we merge this, every non-PR CI build will probably also fail.

Swiftb0y commented:
squashed the fixup commits, rebased on latest 2.3 and fixed pre-commit

Holzhaus commented:
Yup
